examples/src/Auction/Offchain.hs

```diff
@@ -49,18 +50,28 @@ txOffer lot minBid = do
   return $
     -- Debug.trace (show tx) $
     head $ filter (isJust . sBelongsToScript) outputs
+  where
+    seller = walletPKHash sellerWallet
+
+checkWallet :: Monad m => L.PubKeyHash -> Wallet -> m ()
+checkWallet pkh w
+  | pkh /= walletPkh = error $ "wrong wallet arg: got " <> show walletPkh <> "; expected " <> show w
+  | otherwise = pure ()
+  where
+    walletPkh = walletPKHash w
 
 -- | Start an auction by setting the bidding deadline. This transaction consumes
 -- the provided 'Offer' Utxo and returns a 'NoBids' UTxO to the auction
 -- validator. It also mints the thread NFT that ensures the authenticity of the
 -- auction from that point on.
-txSetDeadline :: MonadBlockChain m => SpendableOut -> L.POSIXTime -> m Pl.CardanoTx
-txSetDeadline offerUtxo deadline = do
+txSetDeadline :: MonadWalletMockChain m => SpendableOut -> L.POSIXTime -> Wallet -> m Pl.CardanoTx
+txSetDeadline offerUtxo deadline sellerWallet = do
   let lot = offerUtxo ^. spOutValue
       offerOref = offerUtxo ^. spOutTxOutRef
       theNft = A.threadToken offerOref
   (A.Offer seller minBid) <- spOutGetDatum @A.Auction offerUtxo
-  validateTxSkel $
+  checkWallet seller sellerWallet -- TODO is there a better way?
```

Comment:
I do not understand why this is required. In an audit we might prefer the endpoint not to check they match in order to write traces where they don't and see how the validator handles it.
Also, if we want to ensure they match, we could remove `sellerWallet` from the endpoint parameters and use the retrieved `seller` instead.